hazelcast/src/hazelcast/client/spi.cpp

```diff
@@ -339,29 +345,59 @@ namespace hazelcast {
                         util::hz_snprintf(msg, 100, "(%s:%s) LifecycleService::LifecycleEvent STARTING", date.c_str(),
                                           commitId.c_str());
                         logger.info(msg);
+
+                        for (auto &item: listeners) {
+                            auto& listener = item.second;
+                            listener.starting();
+                        }
                         break;
                     }
-                    case LifecycleEvent::STARTED :
+                    case LifecycleEvent::STARTED : {
                         logger.info("LifecycleService::LifecycleEvent STARTED");
+
+                        for (auto &item: listeners) {
+                            auto& listener = item.second;
```

Comment:
this code looks like repeating, can you refactor, and pass the function pointer?

Reply:
done.